### PR TITLE
[skip ci] Update CHANGELOG links, add entries for v0.5.0

### PR DESCRIPTION
**Pull request by adrienthebo**
This commit updates the CHANGELOG.md to include links for versions,
links for pull request entries, separates additions and changes for
the unreleased version, catches up pull request
of 2019-02-28.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,44 +1,96 @@
...
 ## [Unreleased]
+
+## [v1.0.0] - Unreleased
+### Added
+* Allow creation of service accounts. [#80]
+* Add support for private clusters via submodule. [#69]
+* Add `remove_default_node_pool` set to `false` by default. Fixes [#15]. [#55]
+* Allow arbitrary key-value pairs to be set on node pool metadata. [#52]
+* Add `initial_node_count` parameter to node_pool block. [#60]
+* Fix permanent metadata skew due to disable-legacy-endpoints keys [#114]
```
Comment (aaron-lane) [inline CHANGELOG.md]: This should move under Fixed. The Added entry should identify the new variable.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,44 +1,96 @@
...
+
+Extending the adopted spec, each change should have a link to its corresponding pull request appended.
 
 ## [Unreleased]
+
+## [v1.0.0] - Unreleased
+### Added
+* Allow creation of service accounts. [#80]
+* Add support for private clusters via submodule. [#69]
+* Add `remove_default_node_pool` set to `false` by default. Fixes [#15]. [#55]
```
Comment (aaron-lane) [inline CHANGELOG.md]: An explanation of what was fixed should go under Fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,44 +1,96 @@
...
-* Add `initial_node_count` parameter to node_pool block. #60
+* Set `horizontal_pod_autoscaling` to `true` by default. Fixes [#42]. [#54]
+* Update simple-zonal example GKE version to supported version. [#49]
+* Remove explicit versions from test cases and examples. [#62]
+* Set up submodule structure for public and private clusters. [#61]
+* Update the google and google-beta providers to v2.2 [#106]
+
+### Fixed
+* Zonal clusters can now accept a single zone. Fixes [#43]. [#50]
+* Drop explicit version from simple_zonal example. [#74]
```
Comment (aaron-lane) [inline CHANGELOG.md]: This should go under Changed.

File: CHANGELOG.md (hunk 2 above)
Comment (adrienthebo) [reply CHANGELOG.md]: #15 was a feature request that #55 implemented; I've updated this line to indicate that it's implementing a feature.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,44 +1,96 @@
...
+[#50]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/50
+[#49]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/49
+[#46]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/46
+[#38]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/38
+[#33]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/33
+[#31]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/31
+[#22]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/22
+[#19]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/19
+[#16]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/16
+[#15]: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/15
```
[the remark this answers is not in the document]
Comment (thefirstofthe300) [reply CHANGELOG.md]: Addressed in incoming replacement PR.

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (thefirstofthe300) [reply CHANGELOG.md]: Addressed in replacement PR.

File: CHANGELOG.md (hunk 1 above)
Comment (thefirstofthe300) [reply CHANGELOG.md]: Fixed in replacement PR.

File: CHANGELOG.md (hunk 3 above)
Comment (thefirstofthe300) [reply CHANGELOG.md]: Fixed in replacement PR.

**Pull request closed without merging**